pgconn/pgconn.go

```diff
@@ -193,6 +193,10 @@ func ConnectConfig(octx context.Context, config *Config) (pgConn *PgConn, err er
 			if _, ok := cerr.err.(*NotPreferredError); ok {
 				fallbackConfig = fc
 			}
+			if _, ok := cerr.err.(*errTimeout); ok {
+				// once we reach timeout it's useless to check other fallbacks
+				break
+			}
```

Comment:
What I'm saying is it's not always useless to check other fallbacks. `errTimeout` could come from a TCP timeout. If the context has not expired the next fallback should be tried.

Reply:
Got it.
I made one more attempt :D
So now I do a bit different thing, I try to keep track on fallback error and override result error only if needed.
For a deadline errors we should keep only first deadline error in result, every next one is false-positive errors (since we not actually do anything).